index.html

```diff
@@ -525,5 +525,52 @@ <h2>Privacy Considerations</h2>
     <section>
       <h2>Security Considerations</h2>
     </section>
+
+    <section class="appendix informative">
+<h2>IANA Considerations</h2>
+<section id="media-type">
+  <h2>The <code>application/credential-1.1+json</code> Media Type</h2>
```

Comment:
```suggestion
  <h2>The <code>application/credential-1.1+jose+json</code> Media Type</h2>
```

If this PR is about JWT claim sets, the media type identifier should contain that information in there somewhere. 

Otherwise, people might misunderstand this media type to be THE media type for v1.1 VCs.

IOW, this media type could be seen as squatting on the main spec's media type.

Reply:
its a payload type, not a "jose" type... jose currently refers to the entire token....

- https://www.iana.org/assignments/media-types/application/jose+json
- https://www.iana.org/assignments/media-types/application/jose

Reply:
Hmm, I'm confused by this. This media type is to identify a specific JWT claims set that has the `vc` claim in it, right? For example:

```js
{
  "vc": {
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://www.w3.org/2018/credentials/examples/v1"
    ],
    "id": "http://example.edu/credentials/3732",
    "type": [
      "VerifiableCredential",
      "UniversityDegreeCredential"
    ],
    "issuer": "https://example.edu/issuers/565049",
    "issuanceDate": "2010-01-01T00:00:00Z",
    "credentialSubject": {
      "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
      "degree": {
        "type": "BachelorDegree",
        "name": "Bachelor of Science and Arts"
      }
    }
  },
  "iss": "https://example.edu/issuers/565049",
  "nbf": 1262304000,
  "jti": "http://example.edu/credentials/3732",
  "sub": "did:example:ebfeb1f712ebc6f1c276e12ec21"
}
```

Is that right? If so, it seems like we should have `jwt` in the media type somewhere. Perhaps `application/credential-1.1+jwt`? This would be similar to `application/at+jwt` -- where the JWT must have access token claims.

This is to be used in the `typ` claim, right? The `cty` claim is "NOT RECOMMENDED" per [RFC 7519](https://www.rfc-editor.org/rfc/rfc7519.html#section-5.2) except for nested signing / encryption (a JWT within a JWT) cases, which I don't think we're talking about here.

Reply:
JWS can sign anything - does not have to be a JWT. `application/credential-1.1+json` would allow signing using JWS a conformant structure which is not necessarily a JWT?

wrt nested JWT, it is a confusing term so I would not rely on it too much. it is defined as
>    A JWT in which nested signing and/or encryption are employed.  In
      Nested JWTs, a JWT is used as the payload or plaintext value of an
      enclosing JWS or JWE structure, respectively.

so JWS(JWT claims) and JWE(JWT claims) are nested JWTs (though it is usually used for JWE(JWS(JWT claims)) ) is also a nested JWT so it is kind of in scope.

Reply:
> `application/credential-1.1+json` would allow signing using JWS a conformant structure which is not necessarily a JWT?

But this PR says that when `application/credential-1.1+json` is used then it means the payload *is* a JWT Claims set ... right?

If not, when one sees `application/credential-1.1+json` in the `cty` field, what is the expected payload? What does the JSON look like if it isn't a JWT Claims set? It seems we need to enumerate the possibilities in this PR / change the text if it can be something other than that.

Reply:
FWIW JWS(JWT claims) and JWE(JWT claims) are just JWTs *not* nested.

Reply:
Here are some links I read preparing this PR:

- https://stackoverflow.com/questions/32071006/does-http-content-negotiation-respect-media-type-parameters
- https://jsonapi.org/format/1.1/#content-negotiation-all
- https://github.com/OAI/OpenAPI-Specification/issues/2342
- https://www.alvestrand.no/pipermail/ietf-types/2005-March/001560.html

The intention of this PR is to register a media type that describes the normative requirements associated with a `vc-jwt` that is conformant to `1.1`.

See: https://www.w3.org/TR/vc-data-model/#jwt-encoding

Reply:
It seems vital that `claimset` be communicated in the mime type since that's what we're identifying, not a credential.

Perhaps something like `application/vc-1.1-claimset+json` or `application/vcdm-1.1-claimset+json`?

This mime type identifies a JSON `claimset`, as specified in the VC data model 1.1. In other words, it's a JWT Claim Set with the `vc` (or `vp`?) claim, for use according to the VC data model 1.1 spec. In order to construct a credential (or presentation?) from this payload, the mapping rules in the VC data model 1.1 spec must be followed.

Reply:
In my research preparing this PR, I have not seen any example where `claimset` was present in other use of `cty`.

@selfissued noted that would be redundant on the last call and I agree.

Reply:
@msporny, the `application/credential-1.1+jose+json` suggestion doesn't make sense in two ways.

1. There isn't a `+jose` media type suffix, so it's not there for us to use.
2. Even if there was a +jose media type suffix, it would refer to an `application/jose` object, which is different than a JWT Claims Set.  An `application/jose` object is something using either one of the JOSE Compact Serializations or JOSE JSON Serializations.

Can you therefore withdraw the suggestion, @msporny ?  Thanks.

Reply:
> In my research preparing this PR, I have not seen any example where `claimset` was present in other use of `cty`.

Yes, but I think that's because using `cty` when the payload is a claim set is recommended against. This is what I've been saying. If the payload is a claim set, you don't use `cty` -- at least this is what people who follow the RFC have been doing. Using `cty` in that case would be redundancy, IMO. The payload is assumed to be a claim set unless otherwise specified via `cty`, like when the content type is `jwk+json` or a nested JWT.

> @selfissued noted that would be redundant on the last call and I agree.

I don't understand how it's redundant (clearly this is a different redundancy from what I said above). Perhaps there's an assumption about where the media type "may appear" -- but media types don't work that way. So, the media type name should properly reflect what it identifies regardless of where it appears.

Furthermore, using `cty` is a signal that the payload is *not* a claim set, but, rather, is whatever the media type says it is. This, to me, is an even stronger indication that we need to be clear -- we're using `cty` where it is advised against by the RFC, we haven't been able to find cases where other people do it (thus far, unless they are using non-claim sets), and the thing we're identifying is not a credential... but there actually is a thing that *is* a credential that it could be erroneously confused with.

Reply:
> Can you therefore withdraw the suggestion, @msporny ? Thanks.

The suggestion was to use SOMETHING to denote that the object isn't a VC as defined by the v1.1 data model, but a claim set that wraps a VC. So, we don't have to use `+jose` literally, but we do have to differentiate this thing vs. the v1.1 data model because the data structures are observably different. 

One of the data structures is a claims set that *wraps* a VC, the other *is* a VC. Those are two different things.

Reply:
@dlongley wrote:
> Furthermore, using cty is a signal that the payload is not a claim set, but, rather, is whatever the media type says it is. This, to me, is an even stronger indication that we need to be clear -- we're using cty where it is advised against by the RFC

The definition in the PR explicitly says that it is a media type "identifying a JWT Claims Set conforming to the Verifiable Credentials 1.1 JWT format".  So there's no doubt that it's a JWT Claims Set.

As for the JWT spec saying "In the normal case in which nested signing or encryption operations are not employed, the use of this Header Parameter is NOT RECOMMENDED." about `cty`, as the author of that sentence and an author of the JWT BCP, I can definitively say that explicit typing of objects is now favored by the IETF - which wasn't something we thought of a decade ago as we were developing the JWT spec.  Yes, a JWT payload is always a JWT Claims Set, but there are numerous different kinds of JWT Claims Sets - one of which is the one identified by the proposed Media Type.  Using `cty` enables us to use explicit typing to tell the difference between them.

@msporny wrote:
> The suggestion was to use SOMETHING to denote that the object isn't a VC as defined by the v1.1 data model, but a claim set that wraps a VC.

The something that indicates that the object isn't a VC is that it's called a `credential` in the Media Type name.  If we were to define Media Types for Verified Credentials, they would have `vc` (or `verifiable-credential`) in the name - not just `credential` - which indicates a yet-to-be-secured credential.  The credential becomes verifiable once it's secured.

Reply:
@selfissued,

> The definition in the PR explicitly says that it is a media type "identifying a JWT Claims Set conforming to the Verifiable Credentials 1.1 JWT format". So there's no doubt that it's a JWT Claims Set.

Many people will not read the full description, instead only looking at the media type name.

I don't understand your "redundant" argument since it seems that any name at all would be redundant if we're relying on people always reading the description. But, if you are really ok with the name being anything that isn't "redundant", because you think the description is clear, then let's change the name to address my concern and yours. As I said, I don't know what wouldn't be "redundant" to you, so if you could suggest something new to address both of our concerns, please do.

I'm happy for us to use another name that doesn't include "claim set" in it, so long as it won't be confused with a credential or some other thing this group defines / has defined. I don't think it's ideal for the name to not reflect what it is, but I'm more concerned about basic confusion. My preferences are for avoiding confusion first and then second for good naming hygiene (clearly name things after what they are / refer to).

Reply:
@dlongley wrote:
> Many people will not read the full description, instead only looking at the media type name.

I'm not sympathetic to arguments based on the premise that people won't read the specification.  If one follows that to its logical conclusion, we shouldn't bother writing anything down at all.

As already discussed, using the name `credential` aligns with the way we describe the data structures this working group has invented - and is distinct from "Verifiable Credential".  I think we're good to go as-is.